### ISSUE TITLE
New Sails cli

### ISSUE THREAD

**Issue by listepo** [balderdashy/sails#3137]
I propose to create a new Sails CLI and make it a separate module NPM, inherit from Gulp. This will quickly write new features and modules for it.

**Comment by crh3675** [comment]
Hrm, implications on this.  SailsJs creates a quasi-executable that allows us to create new Sails projects in a flash.  Do you really want to manage:

* SailsJS versions
* SailsJS executables

What happens when you run:

 `sails new SomeCoolProject`

and you get an earlier release because you failed to update the executable or path despite the modules installed. For this to work, it would have to learn the current version first and do all sorts of hand-holding and dependency management for older releases.

My idea, let the SailsJS guys do what they are best (forward progress) with and trust that they give the rock-solid solutions that they have been for forward-thinking development.

**Comment by listepo** [comment]
CLI is a development tool. In my understanding correct its architecture should not be in production.

**Comment by sailsbot** [comment]
Thanks for posting, @listepo.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by mikermcneil** [comment]
@crh3675 thanks for the vote of confidence :)

@listepo I appreciate your point here, and it's something I've considered a lot ever since @colinwren [suggested it](https://github.com/balderdashy/sails/pull/283#issuecomment-15939336) years ago.  In reality, it just has never been a priority because it hasn't caused any major issues and doesn't really take that much longer to install.  Hooks, however, are a different story...

A big goal for Sails v1 is providing a clean way to install the smallest-possible version of core without hooks.  And you're right, the CLI is absolutely a development tool that doesn't necessarily need to be installed on the remote machine.  However, even then, it's quite light compared to e.g. the orm hook.  So our current plan is to focus on hooks first.  Depending on where we end up on that timeline-wise, maybe we can also plan on breaking out the sails CLI from the core install down the road.

**Comment by mikermcneil** [comment]
@listepo @crh3675 created the [`sails-cli`](https://www.npmjs.com/package/sails-cli) and [`sails-core`](https://www.npmjs.com/package/sails-core) packages as placeholders (mainly to prevent having to bother isaacs about it later in case anyone tries to [homestead](https://www.npmjs.com/~waterlinejs) the package names -_- )

However, I was also curious which dependencies would actually be pulled out.  The `package.json` files have a list, but it's definitely incomplete (since right now the CLI requires stuff from the `lib/` directory in core).  I'd suggest it's not worth spending any more time exploring that at the moment for the reasons mentioned above.

However, to finish this up, since I can't find more documentation of the approach Colin and I discussed anywhere (I think it may have been mainly over hangouts), here's a summary:

#### Summary

```
npm install sails
|_ sails-cli
|_ sails-core
```

Whether or not these pieces actually end up in different repos (the experience @sgress454 and I have had with generators leads me to believe this might actually be a bad idea) they would be different packages.  This way, both `npm install -g sails` and `npm install sails` would continue to do what they've always done, but you could also install `sails-core` directly (you wouldn't ever install `sails-cli` directly).

#### Problems

Looking back, I don't think this would actually be the best solution.  For one thing, consolidating the  the more we can consolidate
